src/Symfony/Component/Lock/Store/ZookeeperStore.php

```diff
@@ -0,0 +1,246 @@
+     */
+    private $prefix;
+
+    /**
+     * A base node within which all the locks for PHP application are managed in the Zookeeper environment.
+     */
+    const LOCK_NODE = '/PHP_LOCK_NODE';
+
+    /**
+     * Locker constructor.
+     *
+     * @param \Zookeeper $zookeeper  Zookeeper Instance
+     * @param string     $lockPrefix Prefix for a locking resource
+     */
+    public function __construct(
+        Zookeeper $zookeeper,
+        string $lockPrefix = ''
+    ) {
+        $this->zookeeper = $zookeeper;
+        $this->prefix = $lockPrefix;
+    }
+
+    /**
+     * Stores the resource if it's not locked by someone else.
+     *
+     * @param \Symfony\Component\Lock\Key $key the Key that represent the lock resource
+     *
+     * @throws LockConflictedException
+     * @throws LockAcquiringException
+     */
+    public function save(Key $key)
+    {
+        $resource = $this->getKeyResource($key);
+
+        $lockNodePath = self::LOCK_NODE.'/'.$resource; // For example: /PHP_LOCK_NODE/MESSAGE_1234
+        //This makes sure the node dies at the end of a session, thus releasing the lock to avoid starvation
+        $flags = Zookeeper::EPHEMERAL;
+
+        try {
+            $token = $this->getToken($key);
+            $this->createNewLock($lockNodePath, $token, array(), $flags);
+        } catch (LockConflictedException $ex) {
+            throw $ex;
+        } catch (Throwable $throwable) {
+            throw new LockAcquiringException($throwable);
+        }
+    }
+
+    /**
+     * @param \Symfony\Component\Lock\Key $key Key
+     */
+    public function delete(Key $key)
+    {
+        try {
+            $lockNodePath = self::LOCK_NODE.'/'.$this->getKeyResource($key);
+
+            if ($this->exists($key)) {
+                $this->zookeeper->delete($lockNodePath);
+            }
+        } catch (Throwable $throwable) {
+            // For Zookeeper Ephemeral Nodes, the node will be deleted upon session death. But, if we want to unlock
+            // the lock before proceeding further in the session, the client should be aware of this
+            throw new LockReleasingException($throwable);
+        }
+    }
+
+    /**
+     * Returns whether or not the resource exists in the storage.
+     *
+     * @param \Symfony\Component\Lock\Key $key Key
+     *
+     * @return bool
+     */
+    public function exists(Key $key): bool
+    {
+        $lockNodePath = self::LOCK_NODE.'/'.$this->getKeyResource($key);
+        if (!$this->zookeeper->exists($lockNodePath)) {
+            return false;
+        }
+
+        return $this->zookeeper->get($lockNodePath) === $this->getToken($key);
+    }
+
+    /**
+     * Waits until a key becomes free, then stores the resource.
+     *
+     * This store currently does not support this feature.
+     *
+     * @param \Symfony\Component\Lock\Key $key Key
+     *
+     * @throws NotSupportedException
+     */
+    public function waitAndSave(Key $key)
```

Comment:
Can receive `void` as return type and the `@param` can be removed

Reply:
Replaced with `@inheritdoc` instead of removing `@param`.